### ISSUE TITLE
Fix Engagement Ratio Query

### ISSUE THREAD

**Issue by davismtl** [mozilla/fxa-auth-server#1613]
This query is no longer working:
https://sql.telemetry.mozilla.org/queries/1312?p_days=30#2281

Note:
It is slightly pressing since we now link to this query from the Mozilla Data Collective.

Also, as an example of how public facing some of these queries are becoming, I was going to present this dashboard to the other product managers tomorrow. (I'll remove the query in the meantime.)

https://sql.telemetry.mozilla.org/dashboard/fxa-overview?p_days=30

I'm assigning Phil for now since it is his query and we can't edit it.

**Comment by davismtl** [comment]
Here is the error:
![screen shot 2017-01-11 at 2 33 48 pm](https://cloud.githubusercontent.com/assets/19537086/21869154/02156c9c-d80b-11e6-8952-a4b42b609d80.png)

**Comment by rfk** [comment]
Looks like the datasource has accidentally been changed to "presto" instead of the FxA redshift instance:

![screen shot 2017-01-12 at 09 42 48](https://cloud.githubusercontent.com/assets/34695/21869397/8adc3772-d8ab-11e6-9a35-c425f99923cb.png)

**Comment by philbooth** [comment]
> I'm assigning Phil for now since it is his query and we can't edit it.

Huh, no idea how this happened, sorry about that. As far as I know, I haven't even visited redash in the last week or two. But anyway, I'll go and fix it as soon as I'm back on my work machine in the morning, first thing, I promise.

But also, this underlines a wider point that I hold, which is that we should be way more agressive about forking each others queries, so we can depend on our own versions thereof. I don't want to be a blocker on anyone getting value out of some chart I wrote, which is what seems to have happened here. What happens if I'm on holiday or I resigned 6 months ago or I'm dead? 😄

**Comment by philbooth** [comment]
> As an example of how public facing some of these queries are becoming, I was going to present this dashboard to the other product managers tomorrow.

Missed this sentence first time, so I'm extra sorry now, but also it reinforces my previous comment. If you're demoing a chart I think you should demo your own fork. Who knows what changes I could make just beforehand, which then ruin your demo? Whatever you show on screen might be completely different from when you wrote the talk.

Not trying to be difficult on purpose, but it seems like I've been elevated to "blocker" status here without even realising it.

**Comment by davismtl** [comment]
I think there are two parts to this answer...
1) While forking does provide some relief in some cases, it can create a lot of frustration around the appropriate source of truth. If we link to our graphs but then a new forked version is the latest truth, this can be really hard to manage and be bad for accountability. I would be all for forking our queries if we could eventually re-merge them. Without that, improvements and changes never make it back to any of the other forks. There's no master branch. 

This doesn't mean that there isn't a time to do it but if links have been shared, forking doesn't resolve anything. (e.g. graph linked from a PLR)

2) This leads me to the next point, don't feel like a blocker. It's not your fault in any way (and I'm sorry if I sounded like I was sounding the alarm). This is just an unfortunate limitation of the tool and we hope to have this resolved soon. I've been trying to escalate this for some time in the Re:Dash backlog.

Thanks for your help guys.

**Comment by philbooth** [comment]
Fwiw, I tried to update the query but it fails with this error:

<img width="286" alt="Screenshot of redash error saying: It seems like the query has been modified by another user." src="https://cloud.githubusercontent.com/assets/64367/21883380/bf199688-d8a6-11e6-8b72-f0f66bd30989.png" />

Up at the top it shows this:

<img width="753" alt="Screeshot of redash, showing that query was last updated by Allen Short" src="https://cloud.githubusercontent.com/assets/64367/21883419/ee076c86-d8a6-11e6-8ce8-fd77d459c0d3.png" />

No idea who Allen Short is, but if you know him maybe ask him to edit it? It appears that I'm locked out of doing so now.

**Comment by davismtl** [comment]
For some reason, it is loading now. :)
Not sure if you resolved it but I'll close this since it seems resolved.

Also, I'm not sure if I am upset somebody modified our query or happy to know that this is actually possible now. (or will be)

**Comment by arikfr** [comment]
Hi 👋 

(I'm Redash creator)

Just a note on permissions and who can edit stuff in Redash:

1. Until v0.12 only the owner of the query/dashboard and admins could edit them (I think this is how Allen edited this, as he's probably an admin).
2. In v0.12 (which I think you're using) we introduced the ability to share edit permission with others. It's currently disabled by default, but can be enabled with a feature flag.

Re. the error message you got ("It seems like the query...") - this is a message you're supposed to get in case someone changed the query/dashboard since you last loaded it. This was added to prevent people overriding each other's changes, now that an object can have multiple editors.

If you were getting it constantly even after reload it might imply a bug :( So if you happen to get this again, I'll be happy to know.
